### PR TITLE
Input Pipeline rework

### PR DESCRIPTION
**Pull request by M-R-Schaefer**
I have reworked the data pipeline. The NL is no longer precomputed for the entire dataset and stored in memory.
This used up a log of RAM and the conversion of unpadded NLs to tf ragged tensors was terribly slow.
This rework drastically speeds up start up times and reduces memory consumption without compromising performance.

### CODE REVIEW

File: apax/train/run.py [hunk 1]
```diff
@@ -66,15 +67,31 @@ def run(user_config, log_level="error"):
...
+    )
+    ds_stats = compute_scale_shift_parameters(
+        train_ds.inputs,
+        train_ds.labels,
+        config.data.shift_method,
+        config.data.scale_method,
+        config.data.shift_options,
+        config.data.scale_options,
+    )
+    # TODO IMPL DELETE FILES
```
Comment (Tetracarbonylnickel) [inline apax/train/run.py:91]: Which files should be deleted here?

File: apax/data/initialization.py [hunk 2]
```diff
@@ -38,48 +35,48 @@ def load_data_files(data_config):
...
+#         disable_pbar=config.progress_bar.disable_nl_pbar,
+#         pos_unit=config.data.pos_unit,
+#     )
+#     labels = atoms_to_labels(
+#         atoms_list,
+#         additional_properties_info=config.data.additional_properties_info,
+#         read_labels=read_labels,
+#         pos_unit=config.data.pos_unit,
+#         energy_unit=config.data.energy_unit,
+#     )
```
[the remark this answers is not in the document]
Comment (M-R-Schaefer) [reply apax/data/initialization.py:60]: This was part of the old data pipeline. I forgot to remove the comment.

File: apax/data/input_pipeline.py [hunk 3]
```diff
@@ -1,199 +1,83 @@
...
-            The number of structures that are shuffled for choosing the batches. Should be
-            significantly larger than the batch size. It is recommended to use the default
-            value.
-        """
-        self.n_epoch = n_epoch
-        self.batch_size = None
-        self.n_jit_steps = 1
+        if pre_shuffle:
+            shuffle(atoms)
+        self.sample_atoms = atoms[0]
```
[the remark this answers is not in the document]
Comment (M-R-Schaefer) [reply apax/data/input_pipeline.py:50]: yes, I agree

[the remark this answers is not in the document]
Comment (M-R-Schaefer) [reply apax/data/input_pipeline.py:50]: #246

**Review by Tetracarbonylnickel — APPROVED** [review]
If everything is addressed the PR can be merged.

**Pull request merged**